File: tags/Version 0.9.9/COM/Samples/Python/Testpage2PDF.py
# -*- coding: ISO-8859-1 -*-
# Testpage2PDF.py script
# Part of PDFCreator
# License: GPL
# Homepage: http://www.pdfforge.org/products/pdfcreator
# python 2.5, pywin32 2.5.1
# Version: 1.0.0.0
# Date: December, 24. 2007
# Author: Frank Heindörfer
# Comments: Save the test page as pdf-file using
#           the com interface of PDFCreator.

import win32com.client as com
import os

from time import sleep
from win32print import GetDefaultPrinter, SetDefaultPrinter 
from pythoncom import PumpWaitingMessages

sleepTime = 0.25 # in seconds
maxTime = 10     # in seconds
ReadyState = 0

class clsPDFCreatorEvents(object):
	def OneReady(self):
		print "Ready."
		global ReadyState
		ReadyState = 1
	def OneError(self):
		ReadyState = 0
		global PDFCreator
		print "An error is occured!\r\n Error [" + \
		str(PDFCreator.cErrorDetail("Number")) + "]: " + \
		PDFCreator.cErrorDetail("Description")
		errNum = PDFCreator.cErrorDetail("Number")
		PDFCreator = None
		os._exit(errNum)
		
PDFCreator = com.Dispatch('PDFCreator.clsPDFCreator')
PDFCreatorEvents = com.DispatchWithEvents(PDFCreator, clsPDFCreatorEvents)
ProgramIsRunning = PDFCreator.cProgramIsRunning
PDFCreator.cStart("/NoProcessingAtStartup", 1)

options = PDFCreator.cOptions
options.UseAutosave = 1
options.UseAutosaveDirectory = 1
options.AutosaveDirectory = os.getcwd()
options.AutosaveFilename = "Testpage - PDFCreator"
options.AutosaveFormat = 0                            # 0 = PDF
PDFCreator.cOptions = options
PDFCreator.cSaveOptions()
CurrentPrinter = GetDefaultPrinter()
SetDefaultPrinter("PDFCreator")
PDFCreator.cClearCache()
PDFCreator.cPrintPDFCreatorTestpage()
PDFCreator.cPrinterStop = 0

c = 0

while (ReadyState == 0) and (c < (maxTime / sleepTime)):
	c = c + 1
	sleep(sleepTime)
	PumpWaitingMessages() 

SetDefaultPrinter(CurrentPrinter)
sleep(sleepTime)

if ProgramIsRunning == 0:
	PDFCreator.cClose

if ReadyState == 0:
	a = 0
	print "Creating test page as pdf.\r\n\r\n" + \
	"An error is occured: Time is up!"

File: tags/Version 1.3.0/COM/Samples/Ruby/Testpage2PDF.rb
# Testpage2PDF.rb script
# Part of PDFCreator
# License: GPL
# Homepage: http://www.pdfforge.org/products/pdfcreator
# Ruby version: 1.8.6.0
# Version: 1.0.0.1
# Date: December, 4. 2007
# Author: Frank Heindörfer
# Comments: Save the test page as pdf-file using
#           the com interface of PDFCreator.

require 'win32ole'
require 'win32/process'

SleepTime = 10
$readyState = 0

pdfcreator = WIN32OLE.new('PDFCreator.clsPDFCreator')
event = WIN32OLE_EVENT.new(pdfcreator)
event.on_event('eReady') do
 $readyState = 1
end
event.on_event('eError') do
 print "An error is occured!\nError [", 
  pdfcreator.cErrorDetail('Number'), "]: ",
  pdfcreator.cErrorDetail('Description')
 Process.exit
end

pdfcreator.cStart('/NoProcessingAtStartup')
pdfcreator.setproperty('cOption', 'UseAutosave', 1)
pdfcreator.setproperty('cOption', 'UseAutosaveDirectory', 1)
pdfcreator.setproperty('cOption', 'AutosaveDirectory', Dir.getwd.gsub(/\//,'\\'))
pdfcreator.setproperty('cOption', 'AutosaveFilename', 'Testpage - PDFCreator')
pdfcreator.setproperty('cOption', 'AutosaveFormat', 0) # 0 = PDF
DefaultPrinter = pdfcreator.cDefaultprinter
pdfcreator.setproperty('cDefaultprinter', 'PDFCreator')
pdfcreator.cClearCache()
pdfcreator.cPrintPDFCreatorTestpage()
sleep 2
pdfcreator.setproperty('cPrinterStop', false)

1.upto SleepTime do
 pdfcreator.cOption('UseAutosave') # dummy command, otherwise the com server can't send an event to the script
 break if $readyState != 0
 sleep 1
end

pdfcreator.setproperty('cDefaultprinter', DefaultPrinter)
pdfcreator.cClearCache()
pdfcreator.cClose()

if $readyState == 0 then
 print "Creating test page as pdf.\nAn error is occured: Time is up!"
else
 print "Ready"
end
pdfcreator = nil
Process.exit
